src/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilder.java

```diff
@@ -445,16 +445,25 @@ else if (STREAM_TYPES.contains(this.outType))
         return writer;
      }
 
-    private void determineOutputTypeFromFilename() {
-        if (isBCF(this.outFile)) {
+    private void determineOutputTypeFromFilename(final File f) {
```

Comment:
Since this method now has some subtlety to it, I think it needs a good unit test. Recommend that you make this method `protected static` with an `OutputType` return value so that it can be independently tested, and move the assignment to `this.outType` into the constructors for the builder. Then add a quick unit test covering all the cases we care about (file, symlink, stream, etc.).